docs/advanced/stack-config.md

```diff
@@ -268,6 +269,72 @@ volumes:
 ---------------------
 ```
 
+<a name="configuration-variables"></a>
+## Configuration Variables
+
+Variable | Default | Description
+---------|---------|------------
+IMAGE_SSH_AGENT | docksal/ssh-agent:1.0 | The Docker image to use for SSH Agent. This contains the keys that are shared amongst the projects.
+IMAGE_VHOST_PROXY | docksal/vhost-proxy:1.2 | Docker image to use as the VHOST Proxy. This forwards all requests to the appropriate container based upon the hostname that is being requested.
+IMAGE_DNS | docksal/dns:1.0 | Docker image to use for DNS Routing.
+
+Variable | Default | Description
+---------|---------|------------
+DOCKSAL_LOCK_UPDATES | | When set this will allow for Docksal to no longer accept updates. This is usually good in conjunction of `CI=true`.
+DOCKSAL_ENVIRONMENT | local | **Variable is currently not in use**
+DOCKSAL_STATS_OPTOUT | 0 | Allow for collecting of statistical usage of docksal. When set to `1` this will no longer send statistics.
+DOCKER_NATIVE | 0 | Designates whether to use Docker through VirtualBox or Native Docker. On Linux this is always set to `1`. Otherwise it is set to `0`.
+
+### VHost Proxy
+Variable | Default | Description
+---------|---------|------------
+DOCKSAL_DNS_UPSTREAM | | For environments, where access to external DNS servers is blocked, should be set to the LAN DNS server.
+DOCKSAL_VHOST_PROXY_IP | | Used to set the IP address for the VHOST Proxy to listen on. When `CI` variable is set to `true` this will be set to `0.0.0.0`.
+DOCKSAL_DNS_DOMAIN | docksal | This is the domain to use which is tacked on to the end of the projects url.
+DOCKSAL_NO_DNS_RESOLVER | | Allow disabling the DNS resolver configuration (in case there are issues with it). Set to `true` to activate
+
+
+### DB
```

Comment:
@sean-e-dietrich @achekulaev It should probably be noted here that changing some of the DB variables will not make a change until the DB container is rebuild as they are only read then.